src/org/aion/api/impl/Chain.java

```diff
@@ -463,4 +462,51 @@ private static Block toBlock(Message.rsp_getBlock rsp) {
                 .totalDifficulty(new BigInteger(rsp.getTotalDifficulty().toByteArray()))
                 .createBlock();
     }
+
+    /**
+     * GetNonce returns a BigInteger representing the nonce of the account address at the latest
+     * block number.
+     *
+     * @param address
+     *          the class {@link Address Address} of the desired account to get the nonce of.
+     *
+     * @return the account's nonce.
+     */
+    public ApiMsg getNonce(Address address) {
+        if (!apiInst.isInitialized.get()) {
+            return new ApiMsg(-1003);
+        }
+
+        Message.req_getNonce reqBody = Message.req_getNonce.newBuilder()
+            .setAddress(ByteString.copyFrom(address.toBytes())).build();
+
+        byte[] reqHead = ApiUtils.toReqHeader(ApiUtils.PROTOCOL_VER, Message.Servs.s_chain,
+            Message.Funcs.f_getNonce);
+        byte[] reqMsg = ByteUtil.merge(reqHead, reqBody.toByteArray());
+
+        byte[] rsp = this.apiInst.nbProcess(reqMsg);
+        int code = this.apiInst.validRspHeader(rsp);
+        if (code != 1) {
+            return new ApiMsg(code);
+        }
+
+        try {
+            Message.rsp_getNonce resp = Message.rsp_getNonce.
+                parseFrom(ApiUtils.parseBody(rsp).getData());
+
+            byte[] nonce = resp.getNonce().toByteArray();
+            if (nonce == null) {
+                throw new NullPointerException("null nonce");
```

Comment:
use error code instead of throw exception. So you need to define the error code for this message